### ISSUE TITLE
Prysm & Lighthouse client return different Participation rates

### ISSUE THREAD

**Issue by Buttaa** [prysmaticlabs/prysm#6927]
According to @djrtwo the Lighthouse client returns a participation rate of ~75% for Epoch 548. 

Prysm however returns 47.9%. I am not sure which of the two is correct. 

```
curl -X GET "http://localhost:3500/eth/v1alpha1/validators/participation?epoch=548" -H "accept: application/json"
{"epoch":"548","finalized":true,"participation":{"globalParticipationRate":0.47986263,"votedEther":"339712000000000","eligibleEther":"707936000000000"}}
```

**Comment by terencechain** [comment]
@Buttaa can you check what's lighthouse's participation for epoch 547 and 549? I suspect we have a + or - one issue

**Comment by terencechain** [comment]
Prysm:
{"epoch":"547","finalized":true,"participation":{"globalParticipationRate":0.77318144,"votedEther":"547264000000000","eligibleEther":"707808000000000"}}%                                                                                                                                               {"epoch":"548","finalized":true,"participation":{"globalParticipationRate":0.47986263,"votedEther":"339712000000000","eligibleEther":"707936000000000"}}%
{"epoch":"549","finalized":true,"participation":{"globalParticipationRate":0.7221042,"votedEther":"511296000000000","eligibleEther":"708064000000000"}}%

**Comment by djrtwo** [comment]
this is another one of those epochs
https://beaconcha.in/epoch/1408

LH says ~75%

**Comment by djrtwo** [comment]
Lighthouse for those epochs:

* 547
  * `epoch_target_attesting_gwei`: 551488000000000
  * `epoch_active_gwei`: 707808000000000
  * `target_vote`: 0.7791491477914915
* 548
  * `epoch_target_attesting_gwei`: 550912000000000
  * `epoch_active_gwei`: 707936000000000
  * `target_vote`: 0.7781946390634181
* 549
  * `epoch_target_attesting_gwei`: 546624000000000
  * `epoch_active_gwei`: 708064000000000
  * `target_vote`: 0.7719980114791883

I'm not seeing any epoch in that range with a drastic drop in participation

**Comment by terencechain** [comment]
Thanks @Buttaa and @djrtwo, I figured out what the issue was. To get the participation, we do the following steps:
1. retrieve the state that's `requested_epoch + 1`, this requires replay historical blocks starting from an archived state.
2. run `process_epoch` on the retrieved state  
3. parse participations stats of previous epoch from the result of `process_epoch`

The mistake is in the loading of historical blocks for replay. We did not check whether the block is finalized. For participation at 1408, the node will request state at `start_slot(1049)` which is 45088. As we can see below, that block was orphaned. What made it worse was the orphaned block's parent slot is 45059, 29 slots behind. Because of that, the node's state only accounted for 45 attestations vs 1000+ attestations.

https://beaconcha.in/block/45088

After my fix:
{"epoch":"1408","finalized":true,"participation":{"globalParticipationRate":0.75147694,"votedEther":"614624000000000","eligibleEther":"817888000000000"}

Thanks again for reporting this, I'm really glad we caught this one before mainnet!
